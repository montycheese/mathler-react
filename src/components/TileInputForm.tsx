import {useState} from "react";
import {MAX_CHARS_PER_ROW, VALID_CHARS} from "../mathler/Constants";

type TileInputFormProps = {
    onAdd: Function,
    onDelete: Function,
    isNextCharValid: Function,
    currentEquation: string[]
}

export default function TileInputForm({ onAdd, onDelete, currentEquation, isNextCharValid } : TileInputFormProps) {

    const [val, setVal] = useState('');



    const addVal = () => {
        const isValid = isNextCharValid(val);
        if (isValid) {
            onAdd(val);
            setVal('');
        }
    };


    return (
        <>
            <div className="flex w-4/5 sm:w-1/2 m-auto mt-4 gap-x-2 justify-center bg-slate-50	rounded-md p-2 drop-shadow-2xl">
                Input a number 0-9 or the math operation +, -, *, / to create an equation that equals the answer given.
            </div>
          <div className="flex w-full sm:w-1/2 m-auto mt-4 gap-x-2 justify-center">
              <input disabled={currentEquation.length >= MAX_CHARS_PER_ROW}
                     className="rounded-md p-2 drop-shadow-2xl"
                     type="text"
                     value={val}
                     placeholder="Input next value"
                     maxLength={1}
                     onChange={(e) => {
                         if (VALID_CHARS.includes(e.target.value)) {
                             setVal(e.target.value)
                         }
                     }}
                     onKeyDown={(e) => {
                         if (val.length > 0 && e.key === 'Enter') {
                             addVal();
                         }
                     }}
              />
              <button
                  disabled={val.length !== 1 || !VALID_CHARS.includes(val) || currentEquation.length === MAX_CHARS_PER_ROW}
                  className="text-white text-white bg-gray-900 p-2 rounded-lg drop-shadow-md disabled:opacity-25 hover:bg-gray-800 hover:drop-shadow-xl"
                  onClick={addVal}>
                  Add</button>
              <button className="text-white text-white bg-gray-900 p-2 rounded-lg drop-shadow-md disabled:opacity-25 hover:bg-gray-800 hover:drop-shadow-xl"
                      disabled={currentEquation.length === 0}
                      onClick={() => onDelete()}>
                  Delete</button>
          </div>
        </>
    );
}
