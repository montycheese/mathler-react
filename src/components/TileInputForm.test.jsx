import React from 'react';
import renderer from 'react-test-renderer';
import TileInputForm from "./TileInputForm";

it('renders correctly with empty current submissions', () => {
    const noop = () => {};
    const component = renderer.create(
        <TileInputForm onAdd={noop} onDelete={noop} isNextCharValid={noop} currentEquation={[]}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with partial current submissions', () => {
    const noop = () => {};
    const component = renderer.create(
        <TileInputForm onAdd={noop} onDelete={noop} isNextCharValid={noop} currentEquation={['1', '+', "2"]}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with full current submissions', () => {
    const noop = () => {};
    const component = renderer.create(
        <TileInputForm onAdd={noop} onDelete={noop} isNextCharValid={noop} currentEquation={['1', '+', "2", "2", "-", "1"]}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
