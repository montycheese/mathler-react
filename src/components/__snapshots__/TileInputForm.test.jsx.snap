// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with empty current submissions 1`] = `
Array [
  <div
    className="flex w-4/5 sm:w-1/2 m-auto mt-4 gap-x-2 justify-center bg-slate-50	rounded-md p-2 drop-shadow-2xl"
  >
    Input a number 0-9 or the math operation +, -, *, / to create an equation that equals the answer given.
  </div>,
  <div
    className="flex w-full sm:w-1/2 m-auto mt-4 gap-x-2 justify-center"
  >
    <input
      className="rounded-md p-2 drop-shadow-2xl"
      disabled={false}
      maxLength={1}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Input next value"
      type="text"
      value=""
    />
    <button
      className="text-white text-white bg-gray-900 p-2 rounded-lg drop-shadow-md disabled:opacity-25 hover:bg-gray-800 hover:drop-shadow-xl"
      disabled={true}
      onClick={[Function]}
    >
      Add
    </button>
    <button
      className="text-white text-white bg-gray-900 p-2 rounded-lg drop-shadow-md disabled:opacity-25 hover:bg-gray-800 hover:drop-shadow-xl"
      disabled={true}
      onClick={[Function]}
    >
      Delete
    </button>
  </div>,
]
`;

exports[`renders correctly with full current submissions 1`] = `
Array [
  <div
    className="flex w-4/5 sm:w-1/2 m-auto mt-4 gap-x-2 justify-center bg-slate-50	rounded-md p-2 drop-shadow-2xl"
  >
    Input a number 0-9 or the math operation +, -, *, / to create an equation that equals the answer given.
  </div>,
  <div
    className="flex w-full sm:w-1/2 m-auto mt-4 gap-x-2 justify-center"
  >
    <input
      className="rounded-md p-2 drop-shadow-2xl"
      disabled={true}
      maxLength={1}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Input next value"
      type="text"
      value=""
    />
    <button
      className="text-white text-white bg-gray-900 p-2 rounded-lg drop-shadow-md disabled:opacity-25 hover:bg-gray-800 hover:drop-shadow-xl"
      disabled={true}
      onClick={[Function]}
    >
      Add
    </button>
    <button
      className="text-white text-white bg-gray-900 p-2 rounded-lg drop-shadow-md disabled:opacity-25 hover:bg-gray-800 hover:drop-shadow-xl"
      disabled={false}
      onClick={[Function]}
    >
      Delete
    </button>
  </div>,
]
`;

exports[`renders correctly with partial current submissions 1`] = `
Array [
  <div
    className="flex w-4/5 sm:w-1/2 m-auto mt-4 gap-x-2 justify-center bg-slate-50	rounded-md p-2 drop-shadow-2xl"
  >
    Input a number 0-9 or the math operation +, -, *, / to create an equation that equals the answer given.
  </div>,
  <div
    className="flex w-full sm:w-1/2 m-auto mt-4 gap-x-2 justify-center"
  >
    <input
      className="rounded-md p-2 drop-shadow-2xl"
      disabled={false}
      maxLength={1}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Input next value"
      type="text"
      value=""
    />
    <button
      className="text-white text-white bg-gray-900 p-2 rounded-lg drop-shadow-md disabled:opacity-25 hover:bg-gray-800 hover:drop-shadow-xl"
      disabled={true}
      onClick={[Function]}
    >
      Add
    </button>
    <button
      className="text-white text-white bg-gray-900 p-2 rounded-lg drop-shadow-md disabled:opacity-25 hover:bg-gray-800 hover:drop-shadow-xl"
      disabled={false}
      onClick={[Function]}
    >
      Delete
    </button>
  </div>,
]
`;
